### PR TITLE
FindOneAndDelete command implementation

### LINKED ISSUE #319: Implement findOneAndDelete command

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Implement findOneAndDelete command which finds and deletes a document and returns the deleted document in the response
**Which issue(s) this PR fixes**:
Fixes #319 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DeleteOperation.java [hunk 1]
```diff
@@ -138,7 +172,7 @@ private QueryOuterClass.Query buildDeleteQuery() {
    * @return Uni<Boolean> `true` if deleted successfully, else `false` if data changed and no longer
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DeleteOperation.java:172]: Changed to Tuple2, need to update javadoc

File: src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndDeleteCommandResolver.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      CommandContext commandContext, FindOneAndDeleteCommand command) {
+    List<DBFilterBase> filters = resolve(commandContext, command);
+    final SortClause sortClause = command.sortClause();
+    List<FindOperation.OrderBy> orderBy = SortClauseUtil.resolveOrderBy(sortClause);
+    // If orderBy present
+    if (orderBy != null) {
+      return FindOperation.sorted(
+          commandContext,
+          filters,
+          // 24-Mar-2023, tatu: Since we update the document, need to avoid modifications on
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndDeleteCommandResolver.java:62]: Can remove these comments where copying old tests (for all test setup same applies)

**Review by tatu-at-datastax — APPROVED** [review]
Looks good; had minor suggestions but nothing major.

**Review by vkarpov15 — COMMENTED** [review]
LGTM :+1:

**Pull request merged**